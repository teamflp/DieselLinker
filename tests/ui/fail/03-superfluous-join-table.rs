use diesel_linker::relation;

#[derive(Debug)]
struct User;

#[derive(Debug)]
#[relation(
    relation_type = "many_to_one",
    model = "User",
    fk = "user_id",
    join_table = "users_posts", // This is not used for a many_to_one relation
    backend = "sqlite"
)]
struct Post;

fn main() {}
