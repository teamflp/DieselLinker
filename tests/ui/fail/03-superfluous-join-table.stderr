error: `join_table` is only used for `many_to_many` relations.
  --> tests/ui/fail/03-superfluous-join-table.rs:11:5
   |
11 |     join_table = "users_posts", // This is not used for a many_to_one relation
   |     ^^^^^^^^^^

error[E0412]: cannot find type `Post` in this scope
 --> tests/ui/fail/03-superfluous-join-table.rs:6:10
  |
6 | #[derive(Debug)]
  |          ^^^^^ not found in this scope
